tsl/src/continuous_aggs/materialize.c

```diff
@@ -171,15 +175,27 @@ continuous_agg_materialize(int32 materialization_id, bool verbose)
 
 	/*
 	 * Transaction 2: get the values from the invalidation table,
+	 *                and copy to materialization_invalidation_log
+	 */
+	StartTransactionCommand();
+	PushActiveSnapshot(GetTransactionSnapshot());
+	caggs = ts_continuous_aggs_find_by_raw_table_id(cagg_data.raw_hypertable_id);
+	drain_invalidation_log(cagg_data.raw_hypertable_id, &invalidations);
+	copy_invalidation_logs(caggs, invalidations);
+	PopActiveSnapshot();
+	CommitTransactionCommand();
+
+	/*
+	 * Transaction 3:
 	 *                run the materialization, and update completed_threshold
 	 * (eventually this will only read per-cagg log, and a different worker will
 	 *  move from the hypertable log to the per-cagg log)
 	 */
 	StartTransactionCommand();
 	/* we need a snapshot for the SPI commands within continuous_agg_execute_materialization */
 	PushActiveSnapshot(GetTransactionSnapshot());
-
-	drain_invalidation_log(cagg_data.raw_hypertable_id, &invalidations);
+	get_materialization_invalidation_logs(materialization_id, &mat_invalidations);
```

Comment:
why don't we just use &invalidations here. Weird to use another variable just to copy over on the next line.